Packet++/src/LineBasedProtocol.cpp

```diff
@@ -0,0 +1,126 @@
+#include "Logger.h"
+#include "LineBasedProtocol.h"
+
+#include <string.h>
+
+namespace pcpp
+{
+
+    size_t LineBasedProtocolMessage::getOptionOffset() const
+	{
+		size_t maxLen;
+		if (m_DataLen < 5)
+			maxLen = m_DataLen;
+		else
+			maxLen = 5;
+
+		// Find <SP> if exists
+		uint8_t *pos = (uint8_t *)memchr(m_Data, 0x20, maxLen);
+		if (pos)
+			return pos - m_Data;
+
+		// Find Hyphen "-" if exists
+		pos = (uint8_t *)memchr(m_Data, 0x2d, maxLen);
+		if (pos)
+			return pos - m_Data;
+
+		return m_DataLen;
+	}
+
+	void LineBasedProtocolMessage::changeDelimiter(bool toHyphen)
+	{
+		if (toHyphen)
+			memset(&m_Data[getOptionOffset()], 0x2d, 1);
+		else
+			memset(&m_Data[getOptionOffset()], 0x20, 1);
```

Comment:
it'd be nicer to add macros at the top of the file for hyphen (`0x2d`) space (`0x20`)